packages/react-instantsearch-core/src/connectors/connectSmartSort.ts

```diff
@@ -44,8 +44,10 @@ export default createConnector({
     return {
       isVirtualReplica: results.appliedRelevancyStrictness !== undefined,
       isSmartSorted:
-        results.appliedRelevancyStrictness !== undefined &&
-        results.appliedRelevancyStrictness > 0,
+        results.nbHits === results.nbSortedHits
+          ? false
+          : results.appliedRelevancyStrictness !== undefined &&
+            results.appliedRelevancyStrictness > 0,
```

Comment:
```suggestion
         results.nbHits !== results.nbSortedHits &&
           results.appliedRelevancyStrictness !== undefined &&
           results.appliedRelevancyStrictness > 0,
```

this is probably the same, right?